test/integration/data-types.test.js

```diff
@@ -46,15 +46,15 @@ describe(Support.getTestDialectTeaser('DataTypes'), () => {
     await current.sync({ force: true });
 
     await User.create({
-      dateField: moment('2011 10 31', 'YYYY MM DD'),
+      dateField: dayjs('2011 10 31', 'YYYY MM DD'),
```

Comment:
We should not accept custom dayjs objects here, to avoid introducing a breaking change when we remove the dayjs dependency
This should throw instead

Reply:
Okay, I would change that 👍
To be sure, we throw an error if it's another type or instance than : 
 - Date instance
 - string value

Reply:
I think `number` (for timestamp) is supported too, maybe `bigint` but it should be double checked

Reply:
Hi, @ephys after multiple tests, I don't think throwing an error if the time is not a Date, string, number is a good idea.
We should accept any value which can be stringified to a value readable by a Date object.
```js
import dayjs from "dayjs";

console.log(new Date(dayjs().toString()));
console.log(new Date(dayjs()));
// Fri Apr 15 2022 19:40:06 GMT+0200 (heure d’été d’Europe centrale)
```

As we accept custom dayjs objects, we also accept any other custom date object that can be stringified.

```js
import dayjs from "dayjs";
import moment from "moment";

console.log(dayjs(moment()).toDate());
// Fri Apr 15 2022 19:40:06 GMT+0200 (heure d’été d’Europe centrale)
```

We should let the validator validate the type and then let Dayjs parse the string.

I can remove the `dayjs.isDayjs` method if you think it's needed, but it's only an instance checker to avoid changing the timezone if the developer has done the job in his codebase.

Reply:
Yes, actually that sounds fine.

My concern was avoiding a breaking change once we migrate to `Temporal`, but I think the way forward to support temporal will be to introduced [new DataTypes](https://github.com/sequelize/sequelize/issues/14295) instead, so `DataTypes.DATE` will keep using dayjs or other `Date` based libraries.

How does the timeZone option behave if a moment object is passed to the method?

Reply:
Nice 👍 

The timezone is not concerned since it's a config option and used as the second argument of dayjs function.
The moment object is not receive as a moment object, only as a stringified version of the date.